mbedtls/src/x509/certificate.rs

```diff
@@ -253,6 +254,44 @@ impl Certificate {
         }
         result.map(|_| ())
     }
+
+    pub fn verify_callback<F>(
+        chain: &MbedtlsList<Certificate>,
+        trust_ca: &MbedtlsList<Certificate>,
+        err_info: Option<&mut String>,
+        cb: F,
+    ) -> Result<()>
+    where
+        F: VerifyCallback + 'static,
+    {
+        let mut flags = 0;
+        let result = unsafe {
+            x509_crt_verify(
+                chain.inner_ffi_mut(),
+                trust_ca.inner_ffi_mut(),
+                ::core::ptr::null_mut(),
+                ::core::ptr::null(),
+                &mut flags,
+                Some(verify_callback::<F>),
+                &cb as *const _ as *mut c_void,
+            )
+        }
+        .into_result();
+
+        if result.is_err() {
+            if let Some(err_info) = err_info {
+                let verify_info = crate::private::alloc_string_repeat(|buf, size| unsafe {
+                    let prefix = "\0";
+                    x509_crt_verify_info(buf, size, prefix.as_ptr() as *const _, flags)
+                });
+                if let Ok(error_str) = verify_info {
+                    *err_info = error_str;
+                }
+            }
+        }
+        result.map(|_| ())
+    }
```

Comment:
Could we avoid the code duplication between the existing `fn verify()` and this new `fn verify_callback()` by having both these functions call a private helper function that takes an `Option<F>` as argument?